src/server/scripts/Outland/HellfireCitadel/BloodFurnace/instance_blood_furnace.cpp

```diff
@@ -169,28 +170,37 @@ class instance_blood_furnace : public InstanceMapScript
 
             void ResetPrisons()
             {
-                PrisonerCounter5 = PrisonersCell5.size();
                 ResetPrisoners(PrisonersCell5);
+                PrisonerCounter5 = PrisonersCell5.size();
                 HandleGameObject(PrisonCellGUIDs[DATA_PRISON_CELL5 - DATA_PRISON_CELL1], false);
 
-                PrisonerCounter6 = PrisonersCell6.size();
                 ResetPrisoners(PrisonersCell6);
+                PrisonerCounter6 = PrisonersCell6.size();
                 HandleGameObject(PrisonCellGUIDs[DATA_PRISON_CELL6 - DATA_PRISON_CELL1], false);
 
-                PrisonerCounter7 = PrisonersCell7.size();
                 ResetPrisoners(PrisonersCell7);
+                PrisonerCounter7 = PrisonersCell7.size();
                 HandleGameObject(PrisonCellGUIDs[DATA_PRISON_CELL7 - DATA_PRISON_CELL1], false);
 
-                PrisonerCounter8 = PrisonersCell8.size();
                 ResetPrisoners(PrisonersCell8);
+                PrisonerCounter8 = PrisonersCell8.size();
                 HandleGameObject(PrisonCellGUIDs[DATA_PRISON_CELL8 - DATA_PRISON_CELL1], false);
             }
 
-            void ResetPrisoners(GuidSet const& prisoners)
+            void ResetPrisoners(GuidSet& prisoners)
             {
+                std::list<ObjectGuid> deadPrisoners;
                 for (GuidSet::const_iterator i = prisoners.begin(); i != prisoners.end(); ++i)
```

Comment:
why it is not a foreach? '-'

Reply:
According to the blame view (https://github.com/TrinityCore/TrinityCore/blame/3.3.5/src/server/scripts/Outland/HellfireCitadel/BloodFurnace/instance_blood_furnace.cpp#L191), Shauren is currently the last one to have edited that line (see also https://github.com/TrinityCore/TrinityCore/commit/9b316fe689d06af14d800bcb22992da2e6157225#diff-71f0dbd6c5a0bebf11dfc16eda2993aeR210). I presume he will be able to tell you why.

Reply:
@Keader did you read https://github.com/TrinityCore/TrinityCore/pull/23300/files#r286423369 ?

Reply:
Nope, make more sense, i agree with you :)